src/Xamarin.Android.Build.Tasks/Utilities/NativeAssemblyGenerator/Arm32NativeAssemblyGenerator.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Xamarin.Android.Tasks
+{
+	class Arm32NativeAssemblyGenerator : ArmNativeAssemblyGenerator
+	{
+		static readonly NativeType pointer = new NativeType {
+			Size = 4,
+			Alignment = 4,
+			Name = ".long",
+		};
+
+		protected override string ArchName => "armv7-a";
+		protected override string LineCommentStart => "@";
+		protected override string TypeLead => "%";
+		public override bool Is64Bit => false;
+
+		public Arm32NativeAssemblyGenerator (StreamWriter output, string fileName)
+			: base (output, fileName)
+		{}
+
+		protected override NativeType GetPointerType () => pointer;
+
+		public override uint WriteData<T> (TextWriter writer, T value, bool hex = true, string? comment = null, bool useBlockComment = false) where T: struct
+		{
+			Type type = typeof(T);
+			if (type != typeof(long) && type != typeof(ulong)) {
+				return base.WriteData<T> (writer, value, hex, comment, useBlockComment);
+			}
+
+			string nativeTypeName = GetNativeTypeName<T> ();
+			ulong high, low;
+
+			if (type == typeof(long)) {
+				high = (ulong)((((ulong)(long)(object)value) & 0xFFFFFFFF00000000UL) >> 32);
+				low = (ulong)(((ulong)(long)(object)value) & 0xFFFFFFFFUL);
+			} else {
+				high = (((ulong)(object)value) & 0xFFFFFFFF00000000UL) >> 32;
+				low = ((ulong)(object)value) & 0xFFFFFFFFUL;
+			}
+
+			WriteData (writer, nativeTypeName, ToString (low, hex), comment);
+			WriteData (writer, nativeTypeName, ToString (high, hex));
+
+			return GetNativeTypeSize<T> ();
+		}
+
+		protected override void ConfigureTypeMappings (Dictionary<Type, NativeType?> mapping)
+		{
+			base.ConfigureTypeMappings (mapping);
+
+			// Alignments and sizes as per https://github.com/ARM-software/abi-aa/blob/320a56971fdcba282b7001cf4b84abb4fd993131/aapcs32/aapcs32.rst#fundamental-data-types
+			ConfigureTypeMapping<short>	 (".short", size: 2, alignment: 2);
```

Comment:
Where do the *names* come from? I don't see `.short` or `.long` in the referenced URL.

Reply:
The URL describes alignments and sizes of native architectural types, not directive names of the assembler. Directive names are described in https://sourceware.org/binutils/docs-2.37/as/index.html.

Reply:
Thank you for the link.  Pity it doesn't mention `.xword` though.  And `.long` says "see `.int`." `.int` doesn't mention how big it is.  In short, there's no way to independently verify that these directives are correct without having some knowledge of the target assembler, which isn't me. :-/

Reply:
`.xword` is for 64-bit machines (https://sourceware.org/binutils/docs-2.37/as/AArch64-Directives.html#AArch64-Directives), 32-bit machines would use `.qword` (https://sourceware.org/binutils/docs-2.37/as/Quad.html#Quad) or two `.long` (like ARM does). `.int` is the native word size of the architecture. The documentation is lacking sometimes, so it does take some time to familiarize yourself with how all of this works. There are a lot of historical solutions that are kept because of backward compatibility (for instance comment characters on different platforms, directive names, ambiguities on per-platform basis - eg the `.align` directive and more)